python/summarize_sgfs.py

```diff
@@ -0,0 +1,295 @@
+import itertools
+
+from sgfmill import sgf
+import elo
+import os
+import math
+import argparse
+from typing import List, Dict, Tuple, Set, Sequence
+from dataclasses import dataclass
+import warnings
+
+
+@dataclass
+class Record:
+  win: int = 0
+  lost: int = 0
+  draw: int = 0
+
+
+class GameResultSummary:
+  """
+  Summrize Go games results in sgf or sgfs file format under a directory (and its subdirectory).
+  The sgfs file is a special file format provided by Katago, and each row is a single game in a sgf string.
+  sgf file is not required to be in a single line.
+
+  Public functions include:
+  self.add_games(input_file_dir, search_subdir = False): add files into the results and duplicate files will not be
```

Comment:
I had not thought of duplication checking. Very nice.

Reply:
Thanks :)